app/code/Magento/Catalog/Model/Product.php

```diff
@@ -972,7 +972,7 @@ public function afterSave()
     public function setQty($qty)
     {
         if ($this->getData('qty') != $qty) {
-            $this->setData('qty', $qty);
+            $this->setData('qty', (float)$qty);
```

Comment:
Hi @lewisvoncken 
Could you please consider discussion about setters from this PR https://github.com/magento/magento2/pull/18149

Reply:
fixed